src/cfg.rs

```diff
@@ -0,0 +1,120 @@
+use pwd::Passwd;
+use serde::{Deserialize, Serialize};
+use std::env;
+use std::fs::File;
+use std::collections::BTreeMap;
+
+const CONFIG_ENV: &str = "DOORMAN_CONFIG";
+const DEFAULT_CONFIG: &str = "/etc/doorman.yml";
+
+#[derive(Deserialize, Debug)]
+pub struct Options {
+    pub user: String,
+    pub dosemu_container: String,
+    pub rundir: String,
+    pub sysops: Vec<String>,
+}
+
+fn default_door_nodes() -> i8 {
+    return 1;
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Door {
+    pub path: String,
+    #[serde(default = "default_door_nodes")]
+    pub nodes: i8,
+    pub launch: String,
+    pub configure: Option<String>,
+    pub nightly: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Config {
+    pub doorman: Options,
+    pub doors: BTreeMap<String, Door>,
+}
+
+#[derive(Serialize, Debug)]
+pub struct User {
+    pub uid: u32,
+    pub username: String,
+    pub display_name: String,
+    pub is_sysop: bool,
+}
+
+impl Config {
+    pub fn config_path() -> String {
+        let envvar: Option<String> = match env::var_os(CONFIG_ENV) {
+            Some(val) => match val.to_str() {
+                Some(val_to_str) => Some(String::from(val_to_str)),
+                None => None,
+            },
+            None => None,
+        };
+
+        return String::from(match envvar {
+            Some(val) => val,
+            None => String::from(DEFAULT_CONFIG),
+        });
+    }
+
+    pub fn load() -> Result<Config, String> {
+        let config_path = Config::config_path();
+        let config_file = match File::open(&config_path) {
+            Ok(file) => file,
+            Err(e) => {
+                return Err(format!(
+                    "Couldn't load config from {0}: {1}",
+                    config_path, e
+                ))
+            }
+        };
```

Comment:
```suggestion
         let config_file = File::open(&config_path) else {
            return Err(format!("Couldn't load config from {0}", config_path,));
        };
```
As suggested above, you can use the `?` operator here. Alternatively, you can use the [let-else syntax] (https://rust-lang.github.io/rfcs/3137-let-else.html), although the ?operator is preferable.

Generally, I would [avoid using `match` for unwrapping `Option` and `Result`](https://www.lurklurk.org/effective-rust/transform.html). Rust already offers more compact and idiomatic ways of dealing with errors.